ci/build/upload_build_info.sh

```diff
@@ -7,9 +7,10 @@ readonly PIPELINE_POSTMERGE="0189e759-8c96-4302-b6b5-b4274406bf89"
 readonly PIPELINE_CIV1_BRANCH="0183465b-c6fb-479b-8577-4cfd743b545d"
 if [[
   "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_POSTMERGE}" && 
-  "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_CIV1_BRANCH}"
+  "${BUILDKITE_PIPELINE_ID:-}" != "${PIPELINE_CIV1_BRANCH}" &&
+  "${BUILDKITE_BRANCH:-}" != "master"
 ]]; then
-  echo "Skip build info uploading on non-postmerge pipeline."
+  echo "Skip build info uploading on master branch."
```

Comment:
the original condition is on if it is a post request?

like if it is a release branch commit, which also runs on postmerge pipeline, it will also be uploading.

are we sure not uploading test logs info on release branch is the intended behavior?

Reply:
I probably tighten the condition by mistake when I moved it to rayci, so now 90% of tests do not upload on release branch, and only memory_pressure and window tests go to release branch.

We just need to pick one option for all, but let's upload to release branch only if somebody is using it.